src/app/admin/admin-reports/filtered-items/filtered-items.component.ts

```diff
@@ -0,0 +1,301 @@
+import { DspaceRestService } from 'src/app/core/dspace-rest/dspace-rest.service';
+import { RawRestResponse } from 'src/app/core/dspace-rest/raw-rest-response.model';
+import { MetadataField } from 'src/app/core/metadata/metadata-field.model';
+import { MetadataSchema } from 'src/app/core/metadata/metadata-schema.model';
+import { Collection } from 'src/app/core/shared/collection.model';
+import { Community } from 'src/app/core/shared/community.model';
+import { Item } from 'src/app/core/shared/item.model';
+import { getFirstSucceededRemoteListPayload } from 'src/app/core/shared/operators';
+import { isEmpty } from 'src/app/shared/empty.util';
+import { environment } from 'src/environments/environment';
+import { FiltersComponent } from '../filters-section/filters-section.component';
+import { FilteredItems } from './filtered-items-model';
+import { OptionVO } from './option-vo.model';
+import { PresetQuery } from './preset-query.model';
+import { QueryPredicate } from './query-predicate.model';
+
+@Component({
+  selector: 'ds-report-filtered-items',
+  templateUrl: './filtered-items.component.html',
+  styleUrls: ['./filtered-items.component.scss']
+})
+export class FilteredItemsComponent {
+
+  collections: OptionVO[];
+  presetQueries: PresetQuery[];
+  metadataFields: OptionVO[];
+  metadataFieldsWithAny: OptionVO[];
+  predicates: OptionVO[];
+  pageLimits: OptionVO[];
+
+  queryForm: FormGroup;
+  currentPage = 0;
+  results: FilteredItems = new FilteredItems();
+  results$: Observable<Item[]>;
+  @ViewChild('acc') accordionComponent: NgbAccordion;
+
+  constructor(
+    private communityService: CommunityDataService,
+    private collectionService: CollectionDataService,
+    private metadataSchemaService: MetadataSchemaDataService,
+    private metadataFieldService: MetadataFieldDataService,
+    private translateService: TranslateService,
+    private formBuilder: FormBuilder,
+    private restService: DspaceRestService) {}
+
+  ngOnInit() {
+    this.loadCollections();
+    this.loadPresetQueries();
+    this.loadMetadataFields();
+    this.loadPredicates();
+    this.loadPageLimits();
+
+    let formQueryPredicates: FormGroup[] = [
+      new QueryPredicate().toFormGroup(this.formBuilder)
+    ];
+
+    this.queryForm = this.formBuilder.group({
+      collections: this.formBuilder.control([''], []),
+      presetQuery: this.formBuilder.control('new', []),
+      queryPredicates: this.formBuilder.array(formQueryPredicates),
+      pageLimit: this.formBuilder.control('100', []),
+      filters: FiltersComponent.formGroup(this.formBuilder),
+      additionalFields: this.formBuilder.control([], [])
+    });
+  }
+
+  loadCollections(): void {
+    this.collections = [];
+    let wholeRepo$ = this.translateService.stream('admin.reports.items.wholeRepo');
+    this.collections.push(OptionVO.collectionLoc('', wholeRepo$));
+
+    this.communityService.findAll({ elementsPerPage: 10000, currentPage: 1 }).pipe(
```

Comment:
It's really bad for performance to load a 1000 communities at once. First it'll put a heavy strain on the backend, second it'll take a long time to parse and cache them all on the front end. Since you need colections, not communities I'd try and reuse one of the components that load collections in a paginated way already, such as [ds-dso-selector](https://github.com/DSpace/dspace-angular/blob/d21ce380118240134d7b8838a27bde80cfb95107/src/app/shared/dso-selector/dso-selector/dso-selector.component.ts#L58) or [ds-collection-dropdown](https://github.com/DSpace/dspace-angular/blob/b878f28db6362404433fd4eda4102a7fa09b3cc1/src/app/shared/collection-dropdown/collection-dropdown.component.ts#L51)

Reply:
Basically we agree, but we kept focus on reproducing what was implemented in DSpace 6. Improving performance by various means such as paginating results could be done in a second iteration of the code. Besides, we tried to find our way in the complex code architecture of the REST layer, and we confirm it's not simple, even with more than 24 years experience in Java development!